torch/csrc/jit/tensorexpr/cuda_codegen.cpp

```diff
@@ -469,6 +475,132 @@ std::string CudaCodeGen::GetUniqueFuncName(const std::string& func_prefix) {
   return func_prefix + "_" + std::to_string(value);
 }
 
+// Find all the statements that are not covered by any thread-idx axes,
+// and wrapp them under a trivial thread idx.
+class NoThreadIdxRewriter : public IRMutator {
+ private:
+  Stmt* rewrite(const std::vector<Stmt*>& stmts) {
+    std::vector<Stmt*> cloned_stmts(stmts.size());
+    for (int index = 0; index < stmts.size(); index++) {
+      cloned_stmts[index] = Stmt::clone(stmts[index]);
+    }
+    Stmt* new_block = Block::make(cloned_stmts);
+    // Wrap the new block under a trivial thread-idx
+    //   for t in 0..1: // threadIdx
+    //     if (t < 1):
+    //       new_block
```

Comment:
Not a concern for this PR, but this structure seems like an overkill for just setting a thread-idx for new_block. Is there another way we can represent it?

Reply:
Add a note that this serves too purposes: 1. setting the mask; 2. setting the sync-point boundary.